src/checks.py

```diff
@@ -521,7 +523,7 @@ def check_build_and_test(state: State) -> R:
         print(f"Executing build-and-test for {strategy.name()}")
         err = strategy.run(state)
         if err is not None:
-            errors += f"{strategy.name()}: {err}"
+            errors.append(f"{strategy.name()}: {err}")
```

Comment:
`errors` here became a string, which we then joined using `\n`, and so printed one character per line. Let's not do that.